src/Microsoft.Identity.Client/User.cs

```diff
@@ -47,37 +46,34 @@ internal User(User other)
             IdentityProvider = other.IdentityProvider;
         }
 
+        public User(string identifier, string displayableId, string name, string identityProvider)
+        {
+            if (string.IsNullOrWhiteSpace(identifier))
+            {
+                throw new ArgumentNullException(nameof(identifier));
+            }
+
+            DisplayableId = displayableId;
+            Name = name;
+            IdentityProvider = identityProvider;
```

Comment:
Whole point of unit testing MSAL was to get coverage for API behavior. Developer testing MSAL would either have a failure or have a auth result come back. What additional scenarios do you think the developer need to test to be able to test negative behavior? I don't see a need for user object to have an interface or public constructors